node/sc/multi_bridge_test.go

```diff
@@ -354,3 +356,592 @@ func TestMultiBridgeKLAYTransfer3(t *testing.T) {
+
+	for {
+		select {
+		case _ = <-info.handleCh:
+			t.Fatal("Failure is expected")
+		case err := <-info.handleSub.Err():
+			t.Fatal("Contract Event Loop Running Stop by sub.Err()", "err", err)
+		case <-time.After(timeOut):
+			return
+		}
+	}
+}
+
+// TestMultiBridgeErrOverSign checks the following:
+// - set threshold to 3.
+// - two operators succeed to handle value transfer.
+// - the last operator fails to handle value transfer because of vote closing.
+func TestMultiBridgeErrOverSign(t *testing.T) {
+	info := prepareMultiBridgeEventTest(t)
+	to := common.Address{100}
+	acc := info.accounts[0]
+
+	opts := &bind.TransactOpts{From: acc.From, Signer: acc.Signer, GasLimit: gasLimit}
+	tx, err := info.b.SetOperatorThreshold(opts, voteTypeValueTransfer, 2)
+	assert.NoError(t, err)
+	info.sim.Commit()
+	assert.Nil(t, bind.CheckWaitMined(info.sim, tx))
+
+	nonceOffset := uint64(17)
+	sentNonce := nonceOffset
+	transferAmount := uint64(100)
+	sentBlockNumber := uint64(100000)
+
+	tx = SendHandleKLAYTransfer(info.b, acc, to, transferAmount, sentNonce, sentBlockNumber, t)
+	info.sim.Commit()
+	assert.NoError(t, bind.CheckWaitMined(info.sim, tx))
+
+	acc = info.accounts[1]
+	tx = SendHandleKLAYTransfer(info.b, acc, to, transferAmount, sentNonce, sentBlockNumber, t)
+	info.sim.Commit()
+	assert.NoError(t, bind.CheckWaitMined(info.sim, tx))
+
+	acc = info.accounts[2]
+	tx = SendHandleKLAYTransfer(info.b, acc, to, transferAmount, sentNonce, sentBlockNumber, t)
+	info.sim.Commit()
+	assert.Error(t, bind.CheckWaitMined(info.sim, tx))
+}
+
+// TestMultiBridgeSetKLAYFeeErrNonce checks the following:
+// - failed to set KLAY fee because of the wrong nonce.
+func TestMultiBridgeSetKLAYFeeErrNonce(t *testing.T) {
+	info := prepareMultiBridgeEventTest(t)
+	acc := info.accounts[0]
+	const confThreshold = uint8(2)
+	const fee = 1000
+	requestNonce := uint64(0)
+
+	opts := &bind.TransactOpts{From: acc.From, Signer: acc.Signer, GasLimit: gasLimit}
+	tx, err := info.b.SetOperatorThreshold(opts, voteTypeConfiguration, confThreshold)
+	assert.NoError(t, err)
+	info.sim.Commit()
+	assert.Nil(t, bind.CheckWaitMined(info.sim, tx))
+
+	opts = &bind.TransactOpts{From: acc.From, Signer: acc.Signer, GasLimit: gasLimit}
+	tx, err = info.b.SetKLAYFee(opts, big.NewInt(fee), requestNonce)
+	assert.NoError(t, err)
+	info.sim.Commit()
+	assert.NoError(t, bind.CheckWaitMined(info.sim, tx))
+
+	ret, err := info.b.FeeOfKLAY(nil)
+	assert.NoError(t, err)
+	assert.Equal(t, big.NewInt(0).String(), ret.String())
+
+	acc = info.accounts[1]
+	opts = &bind.TransactOpts{From: acc.From, Signer: acc.Signer, GasLimit: gasLimit}
+	tx, err = info.b.SetKLAYFee(opts, big.NewInt(fee), requestNonce+1)
+	assert.NoError(t, err)
+	info.sim.Commit()
+	assert.Error(t, bind.CheckWaitMined(info.sim, tx))
+}
+
+// TestMultiBridgeKLAYTransferNonceJump checks the following:
+// - set threshold to 2.
+// - first tx is failed since one operator does not vote.
```

Comment:
I thinks this case is not fail case. It succeed voting but not processing.

Reply:
Right.